### PR TITLE
Allow Insecure HTTPS 

### PR DESCRIPTION
**Pull request by kmitul**
## Closes https://github.com/elastic/connectors-py/issues/2491

Updated the verification logic for server certificates to allow for configurable settings. The `verify_certs` option now defaults to `True` but can be set to `False` in the configuration. This change enhances flexibility, allowing users to opt out of certificate verification in environments with varying security requirements.

#### Pre-Review Checklist
- [x] this PR does NOT contain credentials of any kind, such as API keys or username/passwords (double check `config.yml.example`)
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] this PR has a thorough description
- [x] Tested the changes locally
- [x] Considered corresponding documentation changes
- [x] Contributed any configuration settings changes to the configuration reference

#### Changes Requiring Extra Attention

- [x] Security-related changes (encryption, TLS, SSRF, etc)

### CODE REVIEW

File: connectors/es/client.py [hunk 1]
```diff
@@ -87,11 +87,17 @@ def __init__(self, config):
...
+            options["verify_certs"] = config.get("verify_certs", True)
+            if options["verify_certs"]:
+                if "ca_certs" in config:
+                    ca_certs = config["ca_certs"]
+                    logger.debug(f"Verifying cert with {ca_certs}")
+                    options["ca_certs"] = ca_certs
+                else:
+                    logger.debug("Verifying cert with system certificates")        
+        else:
+            options["verify_certs"] = False
```
Comment (jedrazb) [inline connectors/es/client.py:99]: I don't think we need to explicitly set this to `false` here, es py client should skip this config when dealing with `http` traffic anyway.

**Review by jedrazb — COMMENTED** [review]
Great job  @kmitul  thank you for looking into this 👍  I’ve left a couple of small suggestions. The main concern is highlighting that we don’t want this setting to be used in production, so the comments should be adjusted accordingly.

Also, please make sure you sign the contributor agreement, as we can’t merge your PR without it.

**Comment by jedrazb** [comment]
buildkite test this

**Comment by jedrazb** [comment]
You need to run `make autoformat` to make our linter happy (this reminds me we should perhaps make this a pre-commit hook)

**Comment by kmitul** [comment]
Hi @jedrazb :)
Thank you for review and suggestions 👍 
I have made the necessary fixes!

**Comment by jedrazb** [comment]
buildkite test this

**Review by jedrazb — APPROVED** [review]
LGTM! Thank you for contributing to the repo :)

**Comment by seanstory** [comment]
Awesome stuff. Make sure to check out our contributor program, to get rewards! https://www.elastic.co/community/contributor/submissions

**Comment by jedrazb** [comment]
buildkite test this

**Comment by jedrazb** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.x|[](https://github.com/elastic/connectors/pull/3025)|
|✅|8.17|[](https://github.com/elastic/connectors/pull/3026)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**